src/core/core.scale.js

```diff
@@ -640,9 +640,33 @@ module.exports = Element.extend({
 		var isHorizontal = me.isHorizontal();
 		var optionTicks = me.options.ticks.minor;
 		var tickCount = ticks.length;
-		var labelRotationRadians = helpers.toRadians(me.labelRotation);
-		var cosRotation = Math.cos(labelRotationRadians);
-		var longestRotatedLabel = me.longestLabelWidth * cosRotation;
+
+		// Calculate space needed by label in axis direction.
+		// Idea from https://github.com/chartjs/Chart.js/pull/5252/files#r207742171
+		var rot = helpers.toRadians(me.labelRotation);
+		var cos = Math.abs(Math.cos(rot));
+		var sin = Math.abs(Math.sin(rot));
+
+		var padding = optionTicks.autoSkipPadding;
+		var w = me.longestLabelWidth + padding || 0;
+
+		// FIXME: Add support for multiline labels
+		var tickFont = parseFontOptions(optionTicks);
+		var h = tickFont.size * 1.2 + padding;
```

Comment:
Multiline labels make this more complicated because by default we pick the longest label and rotate that, but it is possible to have a single character width with a large number of lines and that would influence the number of labels that need to be skipped.

I believe early versions of v2 measured all the labels, but there were performance bottlenecks with the `ctx.measureText` function

Reply:
Was actually easy, by using max number of lines used. Will make a pen for this.

Reply:
Cool! Sort of rotating the widest label and the tallest label and then seeing which needs the most space?